src/main/java/com/cburch/logisim/file/XmlWriter.java

```diff
@@ -106,39 +97,36 @@ static String attrsToString(NamedNodeMap a) {
   }
 
   static int stringCompare(String a, String b) {
-    if (a.equals(b)) return 0;
-    else if (a == null) return -1;
-    else if (b == null) return 1;
-    else return a.compareTo(b);
+    if (a == null) return -1;
+    if (b == null) return 1;
+    return a.compareTo(b);
+  }
+  
+  static final void swap(Node top, Node nodeA, Node nodeB) {
+    final var na = nodeA.getNodeName();
```

Comment:
`na` => `nameNodeA` (and so on)?

Reply:
You're very picky today :-)
I agree with good naming, but does it make sense (looking at the whole file) to only make this partial fix....
A code refactoring is important. Newly added files or code refactored files should adhere, but does it make sense to already refactor parts when the rest of the file is in "old state" in case of a bug-fix?
Made the changes anyways.